ast/parser.go

```diff
@@ -867,6 +867,12 @@ func (p *Parser) parseNumber() *Term {
 		}
 	}
 
+	// Check for leading 0's, parsed by math/big.Float.Parse as decimal 0: https://golang.org/pkg/math/big/#Float.Parse
+	if len(prefix) == 0 && len(p.s.lit) > 1 && p.s.lit[0] == '0' && p.s.lit[1] != '.' {
```

Comment:
💭 What about `-00` (`len(prefix)!=0`) and `0e1`?

Reply:
Good point, I've amended the PR and added a test case for these two values